cloudvolume/cloudvolume.py

```diff
@@ -161,14 +161,17 @@ def __init__(self, cloudpath, mip=0, bounded=True, autocrop=False, fill_missing=
   @classmethod
   def from_numpy(cls, arr, vol_path='file:///tmp/image/'+generate_random_string(),
                   resolution=(4,4,40), voxel_offset=(0,0,0), 
-                  chunk_size=(128,128,64), layer_type=None, mip_num=1):
+                  chunk_size=(128,128,64), layer_type=None, max_mip=0):
```

Comment:
We should be able to pass in starting mip and `compress` as well. For what it's worth, around the rest of the codebase, I usually refer to the `mip_num` / `max_mip` parameter as `num_mips`. Might be good to keep it that way for consistency.

Reply:
It would be good to be able to pass in `voxel_offset` too.

Reply:
1. there exists `voxel_offset` parameter
2. added an encoding parameter
3. have not changed the max_mip parameter since we need to use +1 if using num_mips.

Reply:
How did I miss the voxel_offset... :man_facepalming:

Reply:
Why not add `compress` as well as encoding though? `encoding` is e.g. `fpzip` or `compressed_segmentation` while compress is a bitstream compressor e.g. `gzip` (or potentially something like `lz4` in the future).

Reply:
added. 
But is compress not used anywhere?
I have a question here:
https://github.com/seung-lab/cloud-volume/issues/198